rustls/src/server/hs.rs

```diff
@@ -290,10 +286,11 @@ pub struct ExpectClientHello {
 
 impl ExpectClientHello {
     pub fn new(
-        server_config: &ServerConfig,
+        server_config: &Arc<ServerConfig>,
```

Comment:
I think at this point we should already be doing only moves; the cloning of `server_config` should be in the caller, so that this becomes `Arc<ServerConfig>`. That is, at the state machine layer, we're always moving, never cloning.

Reply:
Done.